pkg/devicerepository/store/bleve/store.go

```diff
@@ -176,71 +188,82 @@ func (s *bleveStore) GetModels(req store.GetModelsRequest) (*store.GetModelsResp
 		models = append(models, pb)
 	}
 	return &store.GetModelsResponse{
-		Count:  uint32(len(result.Hits)),
-		Total:  uint32(result.Total),
-		Offset: uint32(searchRequest.From),
+		Count:  uint32(len(result.Hits)),   //nolint:gosec
+		Total:  uint32(result.Total),       //nolint:gosec
+		Offset: uint32(searchRequest.From), //nolint:gosec
 		Models: models,
 	}, nil
 }
 
-// GetTemplate retrieves an end device template for an end device definition.
-func (s *bleveStore) GetTemplate(req *ttnpb.GetTemplateRequest, _ *store.EndDeviceProfile) (*ttnpb.EndDeviceTemplate, error) {
-	var (
-		profile             *store.EndDeviceProfile
-		endDeviceProfileIds = req.GetEndDeviceProfileIds()
+var (
+	errNoIdentifiers = errors.DefineInvalidArgument(
+		"no_identifiers",
+		"no identifiers provided",
+	)
+	errMissingProfileIdentifier = errors.DefineInvalidArgument(
+		"missing_profile_identifier",
+		"both vendor ID and vendor profile ID must be provided",
 	)
-	if endDeviceProfileIds != nil && req.VersionIds != nil {
+)
+
+// GetTemplate retrieves an end device template for an end device definition.
+func (s *bleveStore) GetTemplate(req *ttnpb.GetTemplateRequest) (*ttnpb.EndDeviceTemplate, error) {
+	profileIDs, versionIDs := req.GetEndDeviceProfileIds(), req.GetVersionIds()
```

Comment:
The bleve index for templates uses only the profileIDs. In case no profileIDs are present, the index can't be used and the template fetch falls back on the remote implementation of the store where the versionIDs are used. I could check only for profileIDs here, in the bleve implementation of the GetTemplates and for versionIDs in the fallback method, in the remote implementation of the method.